### PR TITLE
Makes the abductor baton's sleep less cancerous

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

1. Makes the sleep last one minute instead of two
2. Makes abductors unable to put people to sleep who are already asleep

## Why It's Good For The Game

Being unconscious indefinitely is not an enjoyable experience as a player, nor is it kind to allow players to do this to each other. This will hopefully open up the opportunity for the abducted to roleplay their abduction instead of having to stare at a black screen with a pinhole in it for ten minutes with no IC chat or messages. If it encourages you to alt+tab or stop playing entirely it's a poor mechanic.

In case I need to explain why the baton is overpowered: It can stun for 14 seconds, sleep for two minutes, its sleeps can be restarted indefinitely and it requires no charging. Even if abductors can't kill people they can still make them miserable by keeping them asleep for too long. 

## Changelog
:cl: cacogen
balance: Abductor baton's sleep lasts one minute instead of two now
balance: Abductor baton's sleep can't be done to mobs that are already asleep
/:cl:

### CODE REVIEW

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 1]
```diff
@@ -559,6 +559,9 @@ Congratulations! You are now trained for invasive xenobiology research!"}
 
 /obj/item/melee/baton/abductor/proc/SleepAttack(mob/living/L,mob/living/user)
 	playsound(src, stun_sound, 50, TRUE, -1)
+	if(L.IsSleeping())
+		L.visible_message("<span class='danger'>[user] tries to induce sleep in [L] with [src], but [L.p_they()] is already asleep!</span>")
```
Comment (Jared-Fogle) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm:563]:
```suggestion
		L.visible_message("<span class='danger'>[user] tries to induce sleep in [L] with [src], but [L.p_they()] [L.p_is()] already asleep!</span>")
```

Otherwise you're going to get "they is already asleep!" for gender neutral mobs.

**Review by Mothblocks — APPROVED** [review]

**Comment by optimumtact** [comment]
:-1: from me,  this was intended to stop abductors killing people

**Comment by LemonInTheDark** [comment]
Something being strong and something being abusable are different things, I agree with oranges.
If you want to make abduction a more vocal process do something other then this.

**Comment by moocowswag** [comment]
If this was to stop abductors from killing people maybe make more incentives to keep people alive or make the mechanics that abductors currently have more easy to come across? 

For example, I only see experienced abductors use brainwashing. I'm not saying to shove that ability into the abductor's face but making it more "accessible" or "advertising" that mechanic to them would do alot. It is a rare role with a high skill ceiling and alot of mechanics afterall.

**Comment by optimumtact** [comment]
It's about letting less robust players have a chance to play abductor without needing to crit everyone they face just to cuff/teleport.

2 minutes is not that much time.

You'd be better off adding a wakeup function to the baton so they can wake you up after you've been cuffed/teleported, or make it happen on teleport if you're cuffed

There's a lot of better solutions to the problem than the low effort change proposed by the OP

**Comment by necromanceranne** [comment]
Ultimately, 1 minute is basically not nearly enough time for an agent to get their scientist down and ready to teleport the target aboard the station, but the agent does still have their headset silencer to keep them calling for help, which means they're probably just going to babysit them somewhere and keep up the baton slaps. This is as usual, with a slightly more inconvenient period of the target waking up that could possibly blow their cover. This ultimately just makes it more likely for inexperienced abductors to fail, but this was always an issue with the antag given the skill ceiling involved, but what I actually think the intent of the PR is meant to be. A noob trap.

What this doesn't force is abductors killing people, because if this would be a problem after this change, it is worse before, and you need to justify too the admins why you are putting every target into critical condition as an abductor on all servers, since their restrictions are univeral. If this ISN'T enforced, you may as well can abductors altogether because they are frankly the most overpowered antagonist we have played optimally, and rule/admin oversight is what keeps them remotely sane.

Ultimately, a 2 minute sleep is a more reliable method of killing someone than a 1 minute, but 1 minute sleep is still practically death in any other context. And for the abductor, at a 1 minute sleep and with an instant headset breaking gun, they can still do their thing without stun and crit. Cuffing is quick, the stuns pretty long, and headsets aren't a problem. It's just more room for failure for those who don't know what they're doing.

This is common in cult, their closest comparison, mostly because cultists aren't punished for it both mechanically and policy-wise. Abductors are on both fronts. You cant experiment on a corpse and you can't murderbone or randomly kill people in anything but self defense. If people start critting as an abductor, they're a moron and probably should be offered up.

Not being able to maintain the sleep is more of an issue. Normally you want to keep your target sedated while operating and 2 minutes is generous enough for the scientist more than the agent, especially if they are in a hurry and the agent is sent down early with the expectation that the scientist will be quick about his operation. A few times a sloppy scientist has let me escape from my bonds and gat them down because they didnt use their time properly. This is more of an immense inconvenience than anything, but forces abductors to slow down a lot.

In addition, as I mentioned, this is an antagonist with the strictest admin policies around it. It is already kept in line with ooc rules, so the afforded powers it has for doing it's thing seems fair to me. It isn't fair for the victim, but that is the nature of antags and what you come to expect from being a victim of the, especially since the antag isn't able to directly round remove you and has an obscene number of tools to actually make your round more interesting, like brainwashing and buffing glands.

**Comment by necromanceranne** [comment]
TL;DR for dummies: ided pr that doesn't actually do anything but it does make newer players more prone to failure. Actually skilled players will still do all the things they did before in the role. This ultimately only makes abductors less approachable for new players.

**Comment by Jared-Fogle** [comment]
The comments put forth convince me, would also like to see more detail from the OP.

**Comment by cacogen** [comment]
If they can't get handcuffs on with the baton and headset off in a minute then maybe they need to get good. It's not like they can't endlessly reapply the stun or sleep.

**Comment by optimumtact** [comment]
Guess I don't know what I'm talking about or doing

**Comment by cacogen** [comment]
Yes, that's what I said knowing I ran the risk of you closing this. You don't even play the game. Your opinions are reactionary and ill-informed. Half of the things you do are motivated by spite. I don't know why you need the repo to be interwoven with your ego but I imagine it must be tiring.

**Comment by optimumtact** [comment]
orangenpc

**Comment by interestingusernam3** [comment]
>reactionary 
ok commie

**Comment by Pugie** [comment]
if long stuns and KOs are needed to prevent people from killing others then optimumtact truly is sabotaging the codebase by nerfing stuns?????

**Comment by cacogen** [comment]
@Pugie That's a dumb generalisation of his incorrect point of view on this specific issue and you know it. Stuns make it easier to kill but abductors need to keep their abductees alive to complete their goals. So abductors having longer stuns make them less likely to need to kill whereas most PvP roles have no incentive to keep a subdued enemy alive.

@interestingusernam3 Reactionary is a word that means being reactive rather than proactive. It doesn't have any inherent political meaning. If you think the fact I have it in my vocabulary means I have a political inclination towards a system that probably doesn't work and would probably still be miserable to live under even if it did then so be it. Freedom is good. Extreme inequality is bad.
